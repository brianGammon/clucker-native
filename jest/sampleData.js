const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
