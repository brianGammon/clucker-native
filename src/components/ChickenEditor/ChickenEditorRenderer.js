import React from 'react';
import { View, Text, TextInput } from 'react-native';

type Props = {
  name: string,
  breed: string,
  hatched: string,
  onFieldChanged: (fieldName: string, text: string) => void,
};

const ChickenEditorRenderer = ({
  name,
  breed,
  hatched,
  onFieldChanged,
}: Props) => (
  <View style={{ flex: 1, alignItems: 'center' }}>
    <Text>Name:</Text>
    <TextInput
      value={name}
      onChangeText={text => onFieldChanged('name', text)}
    />
    <Text>Breed:</Text>
    <TextInput
      value={breed}
      onChangeText={text => onFieldChanged('breed', text)}
    />
    <Text>Hatched On:</Text>
    <TextInput
      value={hatched}
      onChangeText={text => onFieldChanged('hatched', text)}
    />
  </View>
);

export default ChickenEditorRenderer;
