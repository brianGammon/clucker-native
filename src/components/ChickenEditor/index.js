import ChickenEditorComponent from './ChickenEditor';

export default ChickenEditorComponent;
