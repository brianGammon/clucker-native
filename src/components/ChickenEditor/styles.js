import { StyleSheet } from 'react-native';
import variables from 'native-base/dist/src/theme/variables/platform';

export default StyleSheet.create({
  submit: {
    marginTop: 20,
  },
  error: {
    color: variables.inputErrorBorderColor,
  },
});
