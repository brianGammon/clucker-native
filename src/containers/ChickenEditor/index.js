import React from 'react';
import { Text, Button, View } from 'react-native';

export interface Props {
  navigation: any;
}

export default class ChickenEditor extends React.Component<Props> {
  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ fontSize: 30 }}>Add or Edit Chicken</Text>
        <Button onPress={() => navigation.goBack()} title="Go Back" />
      </View>
    );
  }
}
