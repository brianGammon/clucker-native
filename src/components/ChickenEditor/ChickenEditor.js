import * as React from 'react';
import { connect } from 'react-redux';
import chickenSelector from '../../selectors/chickenSelector';
import ChickenEditorRenderer from './ChickenEditorRenderer';
import { firebaseUpdateRequested } from '../../redux/actions';
import { type Chicken } from '../../types';
import { metaTypes } from '../../redux/constants';

type Props = {
  navigation: any,
  chicken: Chicken,
  chickenId: string,
};

type State = {
  name: string,
  breed: string,
  hatched: string,
};

class ChickenEditor extends React.Component<Props, State> {
  static navigationOptions = ({ navigation }) => ({
    title: `${navigation.getParam('chickenId', null) ? 'Edit' : 'Add'} Chicken`,
  });

  constructor() {
    super();
    this.state = {};
    this.onFieldChanged = this.onFieldChanged.bind(this);
  }

  componentDidMount() {
    const { chicken, chickenId } = this.props;
    if (chickenId) {
      this.setState({
        name: chicken.name,
        breed: chicken.breed,
        hatched: chicken.hatched,
      });
    }
  }

  onFieldChanged(name, text) {
    this.setState({ [name]: text });
  }

  render() {
    return (
      <ChickenEditorRenderer
        {...this.state}
        onFieldChanged={this.onFieldChanged}
      />
    );
  }
}

const mapStateToProps = ({ chickens }, { navigation }) => {
  const chickenId = navigation.getParam('chickenId', 'NO-ID');
  if (chickenId === 'NO-ID') {
    return {};
  }
  return {
    ...chickenSelector(chickens.data, chickenId),
  };
};

const mapDispatchToProps = dispatch => ({
  saveForm: data => dispatch(firebaseUpdateRequested(data, metaTypes.chickens)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ChickenEditor);
