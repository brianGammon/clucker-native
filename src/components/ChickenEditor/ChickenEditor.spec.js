import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/sampleData';
import ChickenEditor from '.';

describe('ChickenEditor component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);
  let paramReturned = 'chicken1';
  const navigation = {
    getParam: jest.fn((param, defaultId) => {
      if (paramReturned) {
        return paramReturned;
      }
      return defaultId;
    }),
  };

  test('Should render in edit mode', () => {
    const wrapper = shallow(
      <ChickenEditor store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render in add new mode', () => {
    paramReturned = null;
    const wrapper = shallow(
      <ChickenEditor store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
