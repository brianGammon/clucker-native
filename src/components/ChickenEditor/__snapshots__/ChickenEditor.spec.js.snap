// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChickenEditor component: Should render in add new mode 1`] = `
<ChickenEditorRenderer
  onFieldChanged={[Function]}
/>
`;

exports[`ChickenEditor component: Should render in edit mode 1`] = `
<ChickenEditorRenderer
  name="The First Chicken"
  onFieldChanged={[Function]}
/>
`;
